src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -1124,48 +1132,34 @@ void TaskSketcherElements::changeEvent(QEvent *e)
     }
 }
 
-TaskSketcherElements::MultIcon::MultIcon(const char* name)
+/* Settings menu ==================================================*/
+void TaskSketcherElements::createSettingsButtonActions()
 {
-    int hue, sat, val, alp;
-    Normal = Gui::BitmapFactory().iconFromTheme(name);
-    QImage imgConstr(Normal.pixmap(qAsConst(Normal).availableSizes()[0]).toImage());
-    QImage imgExt(imgConstr);
-
-    for(int ix=0 ; ix<imgConstr.width() ; ix++) {
-        for(int iy=0 ; iy<imgConstr.height() ; iy++) {
-            QColor clr(imgConstr.pixelColor(ix,iy));
-            clr.getHsv(&hue, &sat, &val, &alp);
-            if (alp > 127 && hue >= 0) {
-                if (sat > 127 && (hue > 330 || hue < 30)) {
-                    clr.setHsv((hue + 240) % 360, sat, val, alp);
-                    imgConstr.setPixelColor(ix, iy, clr);
-                    clr.setHsv((hue + 300) % 360, sat, val, alp);
-                    imgExt.setPixelColor(ix, iy, clr);
-                }
-                else if (sat < 64 && val > 192)
-                {
-                    clr.setHsv(240, (255-sat), val, alp);
-                    imgConstr.setPixel(ix, iy, clr.rgba());
-                    clr.setHsv(300, (255-sat), val, alp);
-                    imgExt.setPixel(ix, iy, clr.rgba());
-                }
-            }
-        }
+    QAction* action = new QAction(QString::fromLatin1("Extended information"), this);
+
+    action->setCheckable(true);
+
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/Elements");
+    {
+        QSignalBlocker block(this);
+        action->setChecked(hGrp->GetBool("Extended Naming", false));

```

Comment:
Let's have @abdullahtahiriyo 's opinion

Reply:
It is supported. However, I agree we should have avoided it. 

Because this parameter is just controlled by the widget, I think we can correct it without annoying the user.